src/core/server/elasticsearch/client/cluster_client.ts

```diff
@@ -95,12 +95,15 @@ export class ClusterClient implements ICustomClusterClient {
   private getScopedHeaders(request: ScopeableRequest): Headers {
     let scopedHeaders: Headers;
     if (isRealRequest(request)) {
-      const authHeaders = this.getAuthHeaders(request);
       const requestHeaders = ensureRawRequest(request).headers;
-      scopedHeaders = filterHeaders(
-        { ...requestHeaders, ...authHeaders },
-        this.config.requestHeadersWhitelist
-      );
+      const requestIdHeaders =
+        request instanceof KibanaRequest ? { 'x-opaque-id': request.id } : {};
+      const authHeaders = this.getAuthHeaders(request);
+
+      scopedHeaders = filterHeaders({ ...requestHeaders, ...requestIdHeaders, ...authHeaders }, [
+        'x-opaque-id',
+        ...this.config.requestHeadersWhitelist,
+      ]);
```

Comment:
This could be implemented using the new client's [support for X-Opaque-Id](https://www.elastic.co/guide/en/elasticsearch/client/javascript-api/7.x/observability.html#_x_opaque_id_support) directly, however I thought it would be simpler to reason about if all the headers were configured in a single place, as a single pattern.

Let me know if anyone thinks we should do this differently.

Reply:
I also prefer keeping everything in a single place.